pkg/agent/proxy/proxier.go

```diff
@@ -1023,6 +1023,72 @@ func (p *proxier) GetServiceFlowKeys(serviceName, namespace string) ([]string, [
 	return flows, groups, found
 }
 
+func (p *proxier) HandlePacketIn(pktIn *ofctrl.PacketIn) error {
+	if pktIn == nil {
+		return errors.New("empty packetin for Antrea Proxy")
+	}
+	matches := pktIn.GetMatches()
+
+	noEpToSelectRegField := openflow.NoEpToSelectRegMark.GetField()
+	noEpToSelectRegValue := openflow.NoEpToSelectRegMark.GetValue()
+	match := openflow.GetMatchFieldByRegID(matches, noEpToSelectRegField.GetRegID())
+	if match == nil {
+		return fmt.Errorf("error getting match NoEpToSelectRegMark")
+	}
+	regValue, err := openflow.GetInfoInReg(match, noEpToSelectRegField.GetRange().ToNXRange())
+	if err != nil {
+		klog.ErrorS(err, "Received error while unloading NoEpToSelectRegMark from OVS reg")
+		return err
+	}
+	// Filter out the packets that don't have reg mark NoEpToSelectRegMark.
+	if regValue&noEpToSelectRegValue != noEpToSelectRegValue {
+		return nil
+	}
```

Comment:
L1032-L1046 is for https://github.com/antrea-io/antrea/pull/4656/files#r1142851197 @tnqn

Reply:
This introduces another way to dispatch packet-in event to shared handlers, while all others use CustomReasonField. I think it's not easy to understand and maintain in the long term, and perhaps error-prone, for example, imagine what will happen when we use 0b101 as a new endpoint selection mark in the future, `0b101&0b100 == 0b100`.

Reply:
I proposed this solution (https://github.com/antrea-io/antrea/issues/4726) to optimize the packetIn mechnism, do you think we can schedule it in the next release?

Reply:
Does NXAST_CONTROLLER2 require more recent OVS than the documented one?

Reply:
No, it is provided since OVS v2.6

Reply:
then it looks great to me.